bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/handler/VehicleChannelHandler.java

```diff
@@ -448,9 +455,25 @@ protected void updateWindows(DoorsWindows dw) {
     }
 
     protected void updatePosition(Location pos) {
-        updateChannel(CHANNEL_GROUP_LOCATION, GPS, PointType
-                .valueOf(Double.toString(pos.coordinates.latitude) + "," + Double.toString(pos.coordinates.longitude)));
-        updateChannel(CHANNEL_GROUP_LOCATION, HEADING, QuantityType.valueOf(pos.heading, Units.DEGREE_ANGLE));
-        updateChannel(CHANNEL_GROUP_LOCATION, ADDRESS, StringType.valueOf(pos.address.formatted));
+        if (pos.coordinates.latitude < 0) {
+            updateChannel(CHANNEL_GROUP_LOCATION, GPS, UnDefType.UNDEF);
+            updateChannel(CHANNEL_GROUP_LOCATION, HEADING, UnDefType.UNDEF);
+            updateChannel(CHANNEL_GROUP_LOCATION, ADDRESS, UnDefType.UNDEF);
+            updateChannel(CHANNEL_GROUP_LOCATION, HOME_DISTANCE, UnDefType.UNDEF);
+        } else {
+            PointType vehicleLocation = PointType.valueOf(
+                    Double.toString(pos.coordinates.latitude) + "," + Double.toString(pos.coordinates.longitude));
+            updateChannel(CHANNEL_GROUP_LOCATION, GPS, vehicleLocation);
+            updateChannel(CHANNEL_GROUP_LOCATION, HEADING, QuantityType.valueOf(pos.heading, Units.DEGREE_ANGLE));
+            updateChannel(CHANNEL_GROUP_LOCATION, ADDRESS, StringType.valueOf(pos.address.formatted));
+            PointType homeLocation = locationProvider.getLocation();
+            if (homeLocation != null) {
+                vehicleLocation.distanceFrom(vehicleLocation);
```

Comment:
Seems my comment got lost somehow. What is the purpose of this line?

Reply:
Seems to be a leftover from debugging - removed